api/tests/Api/ECampApiTestCase.php

```diff
@@ -130,4 +137,109 @@ protected function getExamplePayload(string $resourceClass, string $operationTyp
 
         return array_diff_key(array_merge(array_diff_key($examples, array_flip($exceptExamples)), $attributes), array_flip($exceptAttributes));
     }
+
+    protected function get(?BaseEntity $entity = null, ?User $user = null) {
```

Comment:
Ich wäre hier jetzt eben vorsichtig wie viel wir verallgemeinern und DRYen. Ich bin der Meinung es ist wertvoll wenn man im test selber sieht wie genau der Request gemacht wird. Das ist auch eine Art Dokumentation, und es ist mMn hilfreicher wenn ein Test mal fehlschlägt. Sonst muss man erst mal die Test-Infrastruktur debuggen.

Die Assertions finde ich okay hier, aber die Requests würde ich explizit in jedem normalen Test schreiben. Sonst könnten wir gleich cucumber einführen, wo alle Tests nur noch aus lesbaren Sätzen bestehen.

Reply:
I see.. da gehen unsere Vorstellungen wie viel akzeptabel ist wohl auseinander :-)

Ich bin da irgendwie einfach zu faul, immer wieder das gleiche zu schreiben. Und ich finde, es macht die Tests nicht übersichtlicher, wenn sie sehr repetitiv sind. Da ist halt unter Umständen auch nicht mehr so offensichtlich, was genau getestet wird, da es in der restlichen Verbosity untergeht. 

Was wäre denn für dich eine akzeptable Abstraktion? Z.B. dass bei get/patch/delete die Entity jeweils aktiv mitgegeben wird, anstatt dass diese direkt von `defaultEntity` gelesen wird?

Also anstatt so:
```php
    public function testGetIsDeniedForInactiveCollaborator() {
        $this->get(user: static::$fixtures['user5inactive']);
        $this->assertResponseStatusCodeSame(404);
    }
```

Eher so?
```php
    public function testGetIsDeniedForInactiveCollaborator() {
        $this->get(user: static::$fixtures['user5inactive'], entity: static::$fixtures['storyboardSection1']);
        $this->assertResponseStatusCodeSame(404);
    }
```

Reply:
Ich überlege mir mal bis zur Sitzung wie ich es mir vorstelle. Bis dahin kannst du diesen Kritikpunkt mal ignorieren.